src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -908,6 +908,26 @@ public static unsafe object IsInstanceOf(MethodTable* pTargetType, object obj)
                 return IsInstanceOfClass(pTargetType, obj);
         }
 
+        [RuntimeExport("RhTypeCast_IsSimpleInstanceOf")]
+        public static unsafe bool IsSimpleInstanceOf(MethodTable* typeHnd, object obj)
+        {
+            if (obj == null)
```

Comment:
The implementation of this should be copy&paste of `IsInstanceOfClass_Helper` with the block that deals with variance deleted.